helix-term/src/commands.rs

```diff
@@ -2714,11 +2739,17 @@ fn command_mode(cx: &mut Context) {
                 return;
             }
 
-            let parts = input.split_whitespace().collect::<Vec<&str>>();
+            let mut parts = input.split_whitespace().collect::<Vec<&str>>();
             if parts.is_empty() {
                 return;
             }
 
+            // If command is numeric, insert goto command.
+            if parts.len() == 1 && parts[0].parse::<usize>().ok().is_some() {
+                parts.insert(0, cmd::GOTO_CMD);
+            }
```

Comment:
Rather than mutating parts, you could just call the goto function directly here

Reply:
I did that first, but it requires the `cmd::goto_line_number` to be `pub(super)`. By mutating the user input, I'm using the same access as any other command. But I can also change it if it's alright to make it `pub(super)` or you have another idea?

Reply:
@archseer do you still think I should change it?

Reply:
I think it's okay to make it `pub(super)` here :)

Reply:
Alright. Did that and rebased on master :)